src/index/base.cpp

```diff
@@ -159,23 +159,20 @@ void BaseIndex::Sync()
                 return;
             }
 
-            {
-                LOCK(cs_main);
```

Comment:
In commit "index: decrease ThreadSync cs_main contention" (0faafb57f8298547949cbc0044ee9e925ed887ba)

Note: This commit introduces a race condition, because it is no longer locking cs_main while calling `NextSyncBlock` and setting `m_synced = true`. As a result, a new block could be connected by another thread after `NextSyncBlock` returns null in this thread, but before `m_synced` is set to true, so the block will never be indexed because BlockConnected notifications are ignored while m_synced is false. This should be fixed in #29867